pkg/manager/chemanager_controller.go

```diff
@@ -89,7 +105,17 @@ func (r *CheReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 		return ctrl.Result{}, err
 	}
 
-	return r.updateStatus(ctx, current, changed, host)
+	res, err := r.updateStatus(ctx, current, changed, host)
+
+	if err == nil {
+		// update the shared map
+		managerAccess.Lock()
+		defer managerAccess.Unlock()
+
+		currentManagers[req.NamespacedName] = *current
```

Comment:
when this procedure happens (I mean whole `Reconcile` method)? Why we now need t remember the manager?

Reply:
The newly added comments on the `GetCurrentManagers()` method should explain this.